packages/1155-contracts/src/nft/ZoraCreator1155Impl.sol

```diff
@@ -454,9 +454,18 @@ contract ZoraCreator1155Impl is
     }
 
     /// @notice Get the creator reward recipient address
-    /// @dev The creator is not enforced to set a funds recipient address, so in that case the reward would be claimable by creator's contract
-    function getCreatorRewardRecipient() public view returns (address payable) {
-        return config.fundsRecipient != address(0) ? config.fundsRecipient : payable(address(this));
+    /// @param tokenId The token id to get the creator reward recipient for
+    /// @dev The creator is not enforced to set a funds recipient address for a specific token or contract-wide,
+    ///      so in the case of both the reward recipient is set to the creator's contract,
+    ///      which can be withdrawn by the creator via the `withdrawRewards` function.
+    function getCreatorRewardRecipient(uint256 tokenId) public view returns (address payable) {
+        if (getRoyalties(tokenId).royaltyRecipient != address(0)) {
+            return payable(getRoyalties(tokenId).royaltyRecipient);
+        } else if (config.fundsRecipient != address(0)) {
+            return payable(config.fundsRecipient);
+        } else {
```

Comment:
could remove the extra else at the end and just return it

Reply:
ur optimizooor arc has arrived